controllers/articles-controller.js

```diff
@@ -0,0 +1,34 @@
+const { selectArticles, getArticleByIdModel } = require("../models/articles-model");
+
+exports.getArticleByIdController = async (req, res, next) => {
+    try {
+
+        const article_id = req.params.article_id;
+
+        if (article_id === undefined || isNaN(article_id)) {
```

Comment:
If we have an invalid article id, psql will likely throw an error here for you which can be picked up in our catch block in the controller and passed to our error handlers.
We should always allow psql to throw an error if it's going to, and only throw a custom error if it doesn't.
Also this shouldn't be a secondary res.status().send() we should utilise Promise.reject to throw an error that our catch block can handle.
Lastly, this is logic that should be handled in the model, not the controller 👍